versions/development/SPEC.md

```diff
@@ -689,6 +691,59 @@ Pair values can also be specified within the [workflow inputs JSON](https://gith
 }
 ```
 
+### Optional literals
+
+Any non-optional value can be stored into an optional variable by assigning it to an optional variable.
+This does not change the value, but it changes the type of the value:
+
+```WDL
+Int? maybe_five = 5
+```
+
+The `None` literal is the value that an optional has when it is not defined.
+
+```WDL
+Int? maybe_five_but_is_not = None
+# maybe_five_but_is_not is an undefined optional
+Int? maybe_five_and_is = 5
+# maybe_five_and_is is a defined optional
+Int certainly_five = 5
+# Certainly five is not an optional
+
+Boolean test_defined = defined(maybe_five_but_is_not) # Evaluates to false
+Boolean test_defined2 = defined(maybe_five_and_is) # Evaluates to true
+Boolean test_is_none = maybe_five_but_is_not == None # Evaluates to true, same as !definedmaybe_five_but_is_not)
+Boolean test_not_none = maybe_five_but_is_not != None # Evaluates to false, same as defined(maybe_five_but_is_not)
+Boolean compare_int_to_none = certainly_five == None # This will cause an error, since None only makes sense when dealing with optionals.
+```
+
+This is useful in a number of cases, for example when an output of a command depends on a certain flag:
+```wdl
+task test {
+  input {
+    File inputFile
+    String outputPath
+    Boolean log
+  }
+  command {
+    test ~{true="--log" false="" log} inputFile outputPath
+  }
+  output {
+    File? logFile = if log then "test.log" else None
+  }
+}
+```
+If `log` is true, the command will evaluate to `test --log inputFile outputPath` and `logFile` will be `test.log`.
+If `log` is false, the command will evaluate to `test inputFile outputPath` and `logFile` will be `None`.
+
+## Document
+
+```
+$document = ($import | $task | $workflow)+
```

Comment:
Can you really have more than one workflow in a single document?

Reply:
also, why are we making these additions in this PR? seems out of place.

Reply:
This looks like an accidental reversal of https://github.com/openwdl/wdl/pull/270 happened during a rebase to me

Reply:
(or maybe the PR changes jut need another rebase on `master` to pick up that missing PR?)

Reply:
I definitely did not put that in :slightly_smiling_face:.  I will try if a rebase fixes it.

Reply:
@rhpvorderman Heads up that I don't think your rebasing did the trick as this line is still here.